### ISSUE TITLE
Function Proxy returns 404 when path is matched

### ISSUE THREAD

**Issue by aaronpowell** [Azure/azure-functions-host#2940]
#### Investigative information

Please provide the following:

- Timestamp: **2018-06-04T11:07:34.1530851Z**
- Function App version (1.0 or 2.0-beta): **1.0**
- Function App name: **dddsydney-api**
- Function name(s) (as appropriate): Function Proxy
- Invocation ID: N/A
- Region: Australia East

#### Repro steps

1. Access the proxy URL: http://dddsydney-api.azurewebsites.net/.well-known/swagger

#### Expected behavior

Return the contents of the endpoint being proxied

#### Actual behavior

Receiving a 404

#### Known workarounds

None known

#### Related information 

- Proxy definition is here: https://github.com/dddsydney/dddapi/blob/master/proxies.json#L32-L40
- Other proxies using the substitution work correctly
- Debugging log from proxy is here: https://gist.github.com/aaronpowell/34cc05af024fc587800a74fc188efe2a

**Comment by tohling** [comment]
@aaronpowell , we are investigating and will update this thread when we have more information.

**Comment by alexkarcher-msft** [comment]
Adding in @safihamid 

This looks like an error with how we handle routing requests to the same function app. I think the runtime is treating this requests for the openAPI endpoint like a request to reach a local function.

I was able to repo the issue by creating a proxy with the backend URL pointing to that function's swagger file.

**Comment by safihamid** [comment]
This is by design. We don't allow proxy reroutes for /admin/* endpoints.

**Comment by aaronpowell** [comment]
@safihamid is that documented somewhere? I'm pretty sure I had it working originally.

Is there any guidance on how to produce API documentation if you want to mask the token in the URL?

**Comment by safihamid** [comment]
@soninaren is there a way to access /admin/host/swagger without passing the token? we discussed this internally and there are several security risks involved to open up proxies rewrites for local /admin/* endpoints

**Comment by alexkarcher-msft** [comment]
Wouldn't anonymous routes still present the same issue of being unable to write a proxy to your OpenAPI definition? It sounds like we either need to move the OpenAPI definition to a different route, or add a single exception for this one route to completely unblock this scenario.

**Comment by aaronpowell** [comment]
@alexkarcher-msft I'm not bothered sharing the path, it's just the access token that I don't really want in the URL to share publicly (it makes the url ugle and less memorisable). If I didn't need the token I wouldn't need a proxy.

**Comment by soninaren** [comment]
While we figure out the right way to handle this scenario you can use the following workaround.

Simply create an httpTrigger function with Authorization level set to Anonymous

```CSHARP
#r "Newtonsoft.Json"
using System.Net;
using System.Text;
using Newtonsoft.Json;

public static HttpResponseMessage Run(HttpRequestMessage req, TraceWriter log,ExecutionContext context)
{
    string swaggerDocumentPath = Path.Combine(context.FunctionAppDirectory, ".azurefunctions", "swagger", "swagger.json");
    string swaggerDocument = File.ReadAllText(swaggerDocumentPath);

    if (!File.Exists(swaggerDocumentPath))
    {
        return req.CreateResponse(HttpStatusCode.NotFound);
    }

     return new HttpResponseMessage(HttpStatusCode.OK)
    {
        Content = new StringContent(swaggerDocument, Encoding.UTF8, "application/json")
    };
}
```

**Comment by aaronpowell** [comment]
Thanks @soninaren, I'd done something similar.

I've created a separate issue to track the feature request of having an anonymous way to expose the API documentation URL.
